triton-vm/src/table/program_table.rs

```diff
@@ -264,6 +278,37 @@ impl ExtProgramTable {
     }
 }
 
+#[derive(Debug, Copy, Clone, Display, EnumCountMacro, EnumIter, PartialEq, Hash)]
+pub enum ProgramTableChallengeId {
+    InstructionEvalRowIndeterminate,
+    AddressWeight,
+    InstructionWeight,
+    NextInstructionWeight,
+}
+
+impl Eq for ProgramTableChallengeId {}
```

Comment:
Can this be `#[derive(..., Eq)]`?

Reply:
I thought you could only derive `PartialEq` and not `Eq` but I might be wrong...

Reply:
Given `PartialEq`, you can [derive `Eq`](https://doc.rust-lang.org/std/cmp/trait.Eq.html).